cmd/oci-runtime-tool/generate.go

```diff
@@ -296,12 +290,11 @@ func setupSpec(g *generate.Generator, context *cli.Context) error {
 		gidMaps = context.StringSlice("gidmappings")
 	}
 
+	// Add default user namespace.
 	if len(uidMaps) > 0 || len(gidMaps) > 0 {
-		needsNewUser = true
+		g.AddOrReplaceLinuxNamespace("user", "")
```

Comment:
This will silently create an invalid config with `--uidmappings=… --linux-namespace-add=user:foo`, because the resulting config with call for tweaking a joined namespace.  I'm pro-tweaking though, and we're already doing something silent about this case, so I'm ok with that ;).  Folks who want to ensure validity can pass their final config through to `validate` as a separate step.